components/invitation_modal/add_to_channels.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import deepFreeze from 'mattermost-redux/utils/deep_freeze';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import CloseCircleIcon from 'components/widgets/icons/close_circle_icon';
+import ChannelsInput from 'components/widgets/inputs/channels_input';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+
+import './add_to_channels.scss';
+
+export type CustomMessageProps = {
+    message: string;
+    open: boolean;
+}
+
+export const defaultInviteChannels = deepFreeze({
+    channels: [],
+    search: '',
+});
+
+export type InviteChannels = {
+    channels: Channel[];
+    search: string;
+}
+
+export const defaultCustomMessage = deepFreeze({
+    message: '',
+    open: false,
+});
+
+export type Props = {
+    customMessage: CustomMessageProps;
+    toggleCustomMessage: () => void;
+    setCustomMessage: (message: string) => void;
+    inviteChannels: InviteChannels;
+    onChannelsChange: (channels: Channel[]) => void;
+    onChannelsInputChange: (search: string) => void;
+    channelsLoader: (value: string, callback?: (channels: Channel[]) => void) => Promise<Channel[]>;
+    currentChannelName: string;
+    titleClass?: string;
+}
+
+const RENDER_TIMEOUT_GUESS = 100;
+
+export default function AddToChannels(props: Props) {
+    useEffect(() => {
+        if (props.customMessage.open) {
+            setTimeout(() => {
```

Comment:
some time ago in a different project, I wrote a custom hook for checking if an element was already rendered in the DOM. it used IntersectionObserver function. Let's go this way, but let's talk about recreating this custom hook again (I don't longer have access to that code :p) for possible future scenarios like this.